athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -265,43 +274,59 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request
     }
 
     /**
-     * Used to split-up the reads required to scan the requested index/indices. This initial implementation supports
-     * a single split solution.
+     * Used to split-up the reads required to scan the requested index by shard. Cluster-health information is
+     * retrieved for shards associated with the specified index. A split will then be generated for each shard that
+     * is primary and active.
      * @param allocator Tool for creating and managing Apache Arrow Blocks.
-     * @param request Provides details of the catalog, database, table, andpartition(s) being queried as well as
-     * any filter predicate.
+     * @param request Provides details of the catalog, domain, and index being queried, as well as any filter predicate.
      * @return A GetSplitsResponse which primarily contains:
-     * 1. A Set<Split> which represent read operations Amazon Athena must perform by calling your read function.
+     * 1. A Set<Split> each containing a domain and endpoint, and the shard to be retrieved by the Record handler.
      * 2. (Optional) A continuation token which allows you to paginate the generation of splits for large queries.
-     * @note A Split is a mostly opaque object to Amazon Athena. Amazon Athena will use the optional SpillLocation and
-     * optional EncryptionKey for pipelined reads but all properties you set on the Split are passed to your read
-     * function to help you perform the read.
-     * @throws RuntimeException when the domain does not exist in the map.
+     * @throws RuntimeException when the domain does not exist in the map, or an error occurs while processing the
+     * cluster/shard health information.
      */
     @Override
     public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest request)
             throws RuntimeException
     {
         logger.debug("doGetSplits: enter - " + request);
 
+        // Create set of splits
+        Set<Split> splits = new LinkedHashSet<>();
```

Comment:
is order important?

Reply:
Making the order deterministic allows me to ensure that the expect **shard** and **endpoint** values in the properties map are compared against the actual map values in the correct split. Otherwise, I could be comparing the expected information against the wrong split and failing the test.

Reply:
I dont think this is a good reason for taking a dependency on ordering. Why can't the UT be based on a set and compare that way? there is no where in the SDK or even in Athena where order is preserved for splits.

Reply:
Changed to HashSet.